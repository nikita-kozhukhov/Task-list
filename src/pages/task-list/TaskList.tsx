import { MainHeader } from 'components/main-header/MainHeader';
import { OpenModalButton } from 'components/open-modal-button/OpenModalButton';
import { TasksContainer } from 'components/tasks-container/TasksContainer';

import styles from './TaskList.module.scss';

export const TaskList = () => {
  return (
    <section className={styles['task-list']}>
      <MainHeader />
      <OpenModalButton />
      <TasksContainer />
    </section>
  );
};
