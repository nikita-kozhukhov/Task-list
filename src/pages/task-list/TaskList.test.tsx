import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import tasksReducer from 'store/slices/tasks';
import { TaskList } from './TaskList';

const store = configureStore({
  reducer: {
    tasks: tasksReducer,
  },
});

describe('TaskList', () => {
  test('TaskList рендерит все дочерние элементы', () => {
    render(
      <Provider store={store}>
        <TaskList />
      </Provider>
    );

    expect(
      screen.getByRole('button', { name: /Создать задачу/i })
    ).toBeInTheDocument();
    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();

    const container = document.querySelector('.tasks-container');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('tasks-container');
  });

  test('TaskList snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TaskList />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
