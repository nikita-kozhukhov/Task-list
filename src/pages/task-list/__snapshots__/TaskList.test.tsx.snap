// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TaskList TaskList snapshot 1`] = `
<DocumentFragment>
  <section
    class="task-list"
  >
    <h1
      class="main-header"
    >
      Создать задачу
    </h1>
    <div
      class="open-modal-button"
    >
      <button
        class="button button--primary"
        type="button"
      >
        Создать задачу
      </button>
    </div>
    <div
      class="tasks-container"
    />
  </section>
</DocumentFragment>
`;
